bn_s_mp_to_decimal_fast.c

```diff
@@ -0,0 +1,245 @@
+#include "tommath_private.h"
+#include <string.h>
+#ifdef BN_S_MP_TO_DECIMAL_FAST_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+
+/* store a bignum as a decimal ASCII string */
+mp_err s_mp_to_decimal_fast_rec(const mp_int *number, mp_int *nL, mp_int *shiftL, mp_int *mL, int precalc_array_index,
+                                int left,
+                                char **result,
+                                size_t *maxlen)
+{
+   mp_int q, nLq, r;
+   mp_err err;
+
+   if (precalc_array_index < 0) {
+      int n = mp_get_i32(number), n_orig = n, digits_to_copy = 0, copy_counter;
+      char *result_str = *result;
+      char sprintf_str[4] = "000";
+
+      while (n) {
+         sprintf_str[2 - digits_to_copy] = mp_s_rmap[n % 10];
+         digits_to_copy++;
+         n /= 10;
+      }
+
+      if (!left && n_orig < 100) {
```

Comment:
@fperrad can you please expand [1] with the rules on how such statements should be embraced as of your usual linting?

[1] https://github.com/libtom/libtommath/wiki/The-Sourcecode:-Formatting-and-Style